File: frontend/src/types/review.ts
export interface Review {
  id: string;
  productId: string;
  userId: string;
  username: string;
  rating: number;
  comment: string;
  createdAt: string;
}

export interface CreateReviewPayload {
  productId: string;
  rating: number;
  comment: string;
}

File: nhom12/src/main/java/nhom12/example/nhom12/model/Review.java
package nhom12.example.nhom12.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;
import org.springframework.data.mongodb.core.index.CompoundIndex;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "reviews")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
@CompoundIndex(def = "{'userId': 1, 'productId': 1}", unique = true)
public class Review extends BaseDocument {

  @Indexed private String productId;

  @Indexed private String userId;

  private String username;

  private int rating;

  private String comment;
}
